### PR TITLE
Update Filesystem documentation

### PR DESCRIPTION
**Pull request by bzsolt**
IoT.js-DCO-1.0-Signed-off-by: Zsolt Borbély zsborbely.u-szeged@partner.samsung.com

### CODE REVIEW

File: docs/api/IoT.js-API-File-System.md [hunk 1]
```diff
@@ -6,297 +6,588 @@ The following shows fs module APIs available for each platform.
...
+
+var fd = fs.openSync('test.txt', 'r');
+// do something
+fs.closeSync(fd);
+```
+
+
+### Class method: fs.exists(path, callback)
+* `path` {string} File path to be checked.
+* `callback` {Function(exists: boolean)}
```
Comment (daeyeon) [inline docs/api/IoT.js-API-File-System.md]: Shouldn't we remove `(exists: boolean)`?

File: docs/api/IoT.js-API-File-System.md [hunk 2]
```diff
@@ -6,297 +6,588 @@ The following shows fs module APIs available for each platform.
...
+
+var result = fs.existsSync('test.txt');
+assert.equal(result, true);
+```
+
+
+### Class method: fs.mkdir(path[, mode], callback)
+* `path` {string} Path of the directory to be created.
+* `mode` {number} Permission mode. **Default:** `0777`
+* `callback` {Function(err: Error | null)}
```
Comment (daeyeon) [inline docs/api/IoT.js-API-File-System.md]: Shouldn't we remove `(err: Error | null)`?

File: docs/api/IoT.js-API-File-System.md [hunk 3]
```diff
@@ -6,297 +6,588 @@ The following shows fs module APIs available for each platform.
...
+```js
+var fs = require('fs');
+
+fs.mkdirSync('testdir')
+```
+
+
+### Class method: fs.open(path, flags[, mode], callback)
+* `path` {string} File path to be opened.
+* `flags` {string} Open flags.
```
Comment (daeyeon) [inline docs/api/IoT.js-API-File-System.md]: no explanation for `flags`. Please refer to https://github.com/nodejs/node/blob/master/doc/api/fs.md#fsopenpath-flags-mode-callback

File: docs/api/IoT.js-API-File-System.md [hunk 4]
```diff
@@ -6,297 +6,588 @@ The following shows fs module APIs available for each platform.
...
 
-Reads data from the file specified by fd asynchronously.
+
+### Class method: fs.read(fd, buffer, offset, length, position, callback)
+* `fd` {integer} File descriptor.
+* `buffer` {Buffer} Buffer that the data will be written to.
+* `offset` {number} Offset of the buffer where to start writing.
+* `length` {number} Number of bytes to read.
+* `position` {number} Specifying where to start read data from the file, if `null`, read from current position.
+* `callback` {Function(err: Error | null, bytesRead: number, buffer: Buffer)}
```
Comment (daeyeon) [inline docs/api/IoT.js-API-File-System.md]: For callback, please follow node.js form like below. I've update this on #873
```
{Function} The callback is given the three arguments, (err: Error | null, bytesRead: number, buffer: Buffer).
```

File: docs/api/IoT.js-API-File-System.md [hunk 5]
```diff
@@ -6,297 +6,588 @@ The following shows fs module APIs available for each platform.
...
 
-### `fs.readFile(path[, options], callback)`
-* `path <String>` - file path to be opened.
-* `options <Object>` - options for the operation.
-  * `encoding <String> | <null>` Default: `null` - encoding of the file.
-  * `flag <String>` Default: `r` - file open flag.
-* `callback <Function(err: null | Error, data: Buffer)>`
+### Class method: fs.readdir(path[, options], callback)
+* `path` {string} Directory path to be checked.
+* `options` {Object} Options for the operation.
```
Comment (daeyeon) [inline docs/api/IoT.js-API-File-System.md]: As you may know, `options` needs a detailed explanation.

**Comment by glistening** [comment]
@bzsolt Could you read my comment in #873?

File: docs/api/IoT.js-API-File-System.md (hunk 1 above)
Comment (bzsolt) [inline docs/api/IoT.js-API-File-System.md]: I think no, we shouldn't. It works like in Node.js, or is there something wrong?

**Comment by bzsolt** [comment]
I've applied the agreed style on the PR.

**Comment by yichoi** [comment]
LGTM

**Pull request merged**